tests_e2e/pipeline/templates/execute-tests.yml

```diff
@@ -26,20 +26,23 @@ jobs:
 
       - bash: $(Build.SourcesDirectory)/tests_e2e/pipeline/scripts/execute_tests.sh
         displayName: "Execute tests"
+        continueOnError: true
         env:
           # Add all KeyVault secrets explicitly as they're not added by default to the environment vars
           AZURE_CLIENT_ID: $(AZURE-CLIENT-ID)
           AZURE_CLIENT_SECRET: $(AZURE-CLIENT-SECRET)
           AZURE_TENANT_ID: $(AZURE-TENANT-ID)
           SUBSCRIPTION_ID: $(SUBSCRIPTION-ID)
 
+      - publish: $(Build.ArtifactStagingDirectory)
+        artifact: 'artifacts'
+        displayName: 'Publish test artifacts'
+
       - task: PublishTestResults@2
+        displayName: 'Publish test results'
         inputs:
           testResultsFormat: 'JUnit'
-          testResultsFiles: '**/*junit.xml'
+          testResultsFiles: 'lisa/agent.junit.xml'
           searchFolder: $(Build.ArtifactStagingDirectory)
-          testRunTitle: 'Publish test results'
+          failTaskOnFailedTests: true
```

Comment:
If the Execute Tests tasks fails, the build is declared as a Partial Success. This forces the build to Fail on test failures

Reply:
how this flag(failTaskOnFailedTests) is related and only checks for execute-tests status?

will this not run if I have test failures? I think we need all the time to know summary of failures and successes.

Reply:
The task will mark itself as Failed, but it won't prevent it from publishing the results

from https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/reference/publish-test-results-v2

failTaskOnFailedTests - Fail if there are test failures
boolean. Default value: false.

Optional. When this boolean's value is true, the task will fail if any of the tests in the results file are marked as failed. The default is false, which will simply publish the results from the results file.

Reply:
(see run 5534 for an example)

Reply:
Thanks for explaining, makes sense. I see few runs with warning sign. what is that mean? we have some warnings in test but no failures?

Reply:
Those are partial success (reported as "[Build partially succeeded] ...- WALinuxAgent - 0eeb2c4e")

The test execution failed (either because we could not execute the tests, or because some test failed), but I forced the Pipeline to continueOnError when the test execution task fails.

Reply:
BTW - In those test runs there are no test results due to a bug in LISA. I reported it this morning and they will fix it.